include/swift/AST/DiagnosticsSIL.def

```diff
@@ -179,7 +179,11 @@ ERROR(assignment_to_immutable_value,none,
 ERROR(missing_return,none,
       "missing return in a %select{function|closure}1 expected to return %0",
       (Type, unsigned))
-ERROR(non_exhaustive_switch,none, 
+ERROR(missing_never_call,none,
+      "%select{function|closure}1 with uninhabited return type %0 is missing "
```

Comment:
This can come later, but I don't know if "uninhabited" is a term we want to expose to the average user, especially when the common case is `Never`. @jckarter?